tt_metal/impl/dispatch/work_executor.hpp

```diff
@@ -122,11 +123,14 @@ class WorkExecutor {
         this->worker_thread = std::thread(&WorkExecutor::run_worker, this);
         this->worker_queue.worker_thread_id = std::hash<std::thread::id>{}(this->worker_thread.get_id());
         // Bind a worker tied to a device to a specific CPU core in round robin fashion. Thread affinity == Better Perf.
+        static int num_online_cores = sysconf(_SC_NPROCESSORS_ONLN);
```

Comment:
It can, but it usually shouldn't be, unless the scheduler or OS disable certain cores explicitly.
https://www.gnu.org/software/libc/manual/html_node/Processor-Resources.html